### PR TITLE
:white_check_mark: Adds basic client tests

### PR DESCRIPTION
**Pull request by jmelberg-okta**
✅ Adds basic client initializing tests

### CODE REVIEW

**Comment by jmelberg-okta** [comment]
@lboyette-okta

File: okta/AppInstanceClient.py [hunk 1]
```diff
@@ -5,8 +5,9 @@
 
 
 class AppInstanceClient(ApiClient):
-    def __init__(self, base_url, api_token):
-        ApiClient.__init__(self, base_url + '/api/v1/apps', api_token)
+    def __init__(self, *args, **kwargs):
+        kwargs['pathname'] = '/api/v1/apps'
+        ApiClient.__init__(self, **kwargs)
```
Comment (lboyette-okta) [inline okta/AppInstanceClient.py]: I don't think we want to completely lose the `*args` in any of these clients

File: okta/framework/ApiClient.py [hunk 2]
```diff
@@ -9,8 +9,16 @@
 class ApiClient(object):
 
     def __init__(self, *args, **kwargs):
-        self.base_url = kwargs['base_url'] or args[0]
-        self.api_token = kwargs['api_token'] or args[1]
+        if 'pathname' not in kwargs:
+            raise ValueError('Invalid url')
```
Comment (lboyette-okta) [inline okta/framework/ApiClient.py]: Let's make this more specific - `A pathname must be provided to create an ApiClient`

File: okta/framework/ApiClient.py [hunk 3]
```diff
@@ -9,8 +9,16 @@
...
+
+        if 'base_url' in kwargs and 'api_token' in kwargs:
```
Comment (lboyette-okta) [inline okta/framework/ApiClient.py]: This isn't equivalent to what we had before, so may (unlikely, but may) break consumers. What if they were passing a `base_url` in the `*args`, but an `api_token` in `*kwargs`? We should split this into two `if`/`else` clauses: a `base_url` check and an `api_token` check.

File: tests/AppInstanceClientTest.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        headers={
+            'x-test-description': test_description
+        }
+    )
+
+
+class AppInstanceClientTest(unittest.TestCase):
+
+    def tests_client_initializer(self):
+        client = build_client('/api/v1/events - initialize app instance client')
```
Comment (lboyette-okta) [inline tests/AppInstanceClientTest.py]: great! I like the idea of adding tests for this. The `build_client` method may be a bit of overkill though, and it only allows for one type of instantiation. If we can change this to two tests (one for the `*args` way to instantiate and one for `**kwargs`, that'd be awesome.

File: okta/framework/ApiClient.py [hunk 5]
```diff
@@ -9,8 +9,19 @@
...
+            raise ValueError('A pathname must be provided to create an ApiClient')
+
+        if 'base_url' in kwargs:
+            self.base_url = kwargs['base_url'] + kwargs['pathname']
+
+        if 'api_token' in kwargs:
+            self.api_token = kwargs['api_token']
+
+        else:
+            self.base_url = args[0] + kwargs['pathname']
```
Comment (lboyette-okta) [inline okta/framework/ApiClient.py]: `new ApiClient(null, 'token', base_url='http://sample.com')` will fail

File: tests/AppInstanceClientTest.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import unittest
+import os
+import json
+
+from okta import AppInstanceClient
+
+config_path = os.path.normpath(os.path.join(os.path.dirname(__file__),
```
Comment (lboyette-okta) [inline tests/AppInstanceClientTest.py]: Let's keep this out until we need it

**Pull request merged**